static/app/views/dashboardsV2/widgetBuilder/yAxisSelector.tsx

```diff
@@ -0,0 +1,261 @@
+  }
+
+  const filterAggregateParameters = fieldValue => option => {
+    // Only validate function parameters for timeseries widgets and
+    // world map widgets.
+    if (doNotValidateYAxis) {
+      return true;
+    }
+
+    if (fieldValue.kind !== 'function') {
+      return true;
+    }
+
+    // if (isMetricWidget) {
+    //   return true;
+    // }
+
+    const functionName = fieldValue.function[0];
+    const primaryOutput = aggregateFunctionOutputType(
+      functionName as string,
+      option.value.meta.name
+    );
+    if (primaryOutput) {
+      return isLegalYAxisType(primaryOutput);
+    }
+
+    if (option.value.kind === FieldValueKind.FUNCTION) {
+      // Functions are not legal options as an aggregate/function parameter.
+      return false;
+    }
+
+    return isLegalYAxisType(option.value.meta.dataType);
+  };
+
+  const canDelete = fields.length > 1;
+
+  const hideAddYAxisButtons =
+    ([DisplayType.WORLD_MAP, DisplayType.BIG_NUMBER].includes(displayType) &&
+      fields.length === 1) ||
+    ([
+      DisplayType.LINE,
+      DisplayType.AREA,
+      DisplayType.STACKED_AREA,
+      DisplayType.BAR,
+    ].includes(displayType) &&
+      fields.length === 3);
+
+  let fieldContents: React.ReactElement;
+  if (displayType === DisplayType.TOP_N) {
+    const fieldValue = fields[fields.length - 1];
+    fieldContents = (
+      <QueryFieldWrapper key={`${fieldValue}:0`}>
+        <QueryField
+          fieldValue={fieldValue}
+          fieldOptions={generateFieldOptions({organization})}
+          onChange={value => handleTopNChangeField(value)}
+          filterPrimaryOptions={filterPrimaryOptions}
+          filterAggregateParameters={filterAggregateParameters(fieldValue)}
+        />
+      </QueryFieldWrapper>
+    );
+  } else {
+    fieldContents = (
+      <React.Fragment>
+        {fields.map((fieldValue, i) => (
+          <QueryFieldWrapper key={`${fieldValue}:${i}`}>
+            <QueryField
+              fieldValue={fieldValue}
+              fieldOptions={fieldOptions}
+              onChange={value => handleChangeField(value, i)}
+              filterPrimaryOptions={filterPrimaryOptions}
+              filterAggregateParameters={filterAggregateParameters(fieldValue)}
+              otherColumns={fields}
+            />
+            {(canDelete || fieldValue.kind === 'equation') && (
+              <DeleteButton onDelete={event => handleRemove(event, i)} />
+            )}
+          </QueryFieldWrapper>
+        ))}
+        {!hideAddYAxisButtons && (
+          <Actions>
+            <AddButton title={t('Add Overlay')} onAdd={handleAdd} />
+            <AddButton title={t('Add an Equation')} onAdd={handleAddEquation} />
+          </Actions>
+        )}
+      </React.Fragment>
+    );
+  }
+
+  return (
+    <Field
+      data-test-id="y-axis"
+      label={t('Y-Axis')}
+      inline={false}
+      style={{padding: `${space(2)} 0 24px 0`, ...(style ?? {})}}
```

Comment:
should we use emotion to style this component?

Reply:
Nope! Looks better without the spacing, thanks for catching that.